src/clarity-angular/datagrid/render/row-renderer.ts

```diff
@@ -3,44 +3,132 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import {Directive, ContentChildren, QueryList, OnDestroy, AfterViewInit, AfterContentInit} from "@angular/core";
-import {Subscription} from "rxjs/Subscription";
-import {DatagridRenderOrganizer} from "./render-organizer";
-import {DatagridCellRenderer} from "./cell-renderer";
+import { Directive, ContentChildren, QueryList, OnDestroy, AfterViewInit, AfterContentInit } from "@angular/core";
+import { Subscription } from "rxjs/Subscription";
+import { DatagridRenderOrganizer } from "./render-organizer";
+import { DatagridCellRenderer } from "./cell-renderer";
+import { HideableColumnService } from "../providers/hideable-column.service";
+import { DatagridCell } from "../datagrid-cell";
+import { DatagridHideableColumn } from "../datagrid-hideable-column";
 
 @Directive({
     selector: "clr-dg-row, clr-dg-row-detail"
 })
 export class DatagridRowRenderer implements OnDestroy, AfterContentInit, AfterViewInit {
 
-    constructor(private organizer: DatagridRenderOrganizer) {
+    constructor( private organizer: DatagridRenderOrganizer,
+                 private hideableColumnSvc: HideableColumnService ) {
         this.subscription = organizer.alignColumns.subscribe(() => this.setWidths());
+        // console.log("construct");
     }
 
     private subscription: Subscription;
+
     ngOnDestroy() {
         this.subscription.unsubscribe();
     }
 
     @ContentChildren(DatagridCellRenderer) cells: QueryList<DatagridCellRenderer>;
 
+    /*****
+     * property dgCells
+     *
+     * @description
+     * A Query List of the Datagrid cells in this row.
+     *
+     * @type QueryList<DatagridCell>
+     */
+    @ContentChildren(DatagridCell) dgCells: QueryList<DatagridCell>;
```

Comment:
Moved it to the Row. For some reason I started in RowRenderer early on and things just stayed there.